### PR TITLE
Run condition for composable nodes

### PR DESCRIPTION
**Pull request by adityapande-1995**
Signed-off-by: Aditya <aditya050995@gmail.com>

## Summary 
This PR adds the ``condition`` parameter to composable nodes action. Resolves https://github.com/ros2/launch_ros/issues/294. This action is currently supported by ``ComposableNodeContainer`` but not ``ComposableNode`` itself.

## Example usage
One can use the following example file : 
```
import launch
import launch.actions
from launch.conditions import IfCondition, UnlessCondition
from launch.substitutions import LaunchConfiguration
import launch_ros.actions
from launch_ros.actions import ComposableNodeContainer
from launch_ros.descriptions import ComposableNode

def generate_launch_description():
    return launch.LaunchDescription([
        launch.actions.DeclareLaunchArgument(name='flag', default_value='False'),
        ComposableNodeContainer(
            name='my_container',
            namespace='',
            package='rclcpp_components',
            executable='component_container',

            composable_node_descriptions=[
                ComposableNode(
                    package='composition',
                    plugin='composition::Talker',
                    condition=IfCondition(LaunchConfiguration('flag')),
                    name='talker_1'),

                ComposableNode(
                    package='composition',
                    plugin='composition::Talker',
                    condition=UnlessCondition(LaunchConfiguration('flag')),
                    name='talker_2'),

                ComposableNode(
                    package='composition',
                    plugin='composition::Talker',
                    name='talker_3'),
            ],
            output='screen',
        ),
    ]) 
```
or alternatively as an xml, 

```
<launch>
        <arg name="flag" default="false"/>
        <node_container pkg="rclcpp_components" exec="component_container" name="my_container" namespace="">
                <composable_node pkg="composition" plugin="composition::Talker" name="talker_1" if="$(var flag)" />
                <composable_node pkg="composition" plugin="composition::Talker" name="talker_2" unless="$(var flag)" />
                <composable_node pkg="composition" plugin="composition::Talker" name="talker_3" />
        </node_container>
</launch>
```

Running this  as ``ros2 launch test.py  flag:=True`` will make it launch ``talker_1`` and ``talker_3``, and ``False`` will make it launch ``talker_2`` and ``talker_3``.

### CODE REVIEW

File: launch_ros/launch_ros/actions/composable_node_container.py [hunk 1]
```diff
@@ -79,15 +79,21 @@ def execute(self, context: LaunchContext) -> Optional[List[Action]]:
         composable nodes load action if it applies.
         """
         load_actions = None  # type: Optional[List[Action]]
+        valid_composable_nodes = []
+        for node_object in self.__composable_node_descriptions:
+            if node_object.condition() is None:
+                valid_composable_nodes.append(node_object)
+            elif node_object.condition().evaluate(context):
+                valid_composable_nodes.append(node_object)
```
Comment (ivanpauno) [inline launch_ros/launch_ros/actions/composable_node_container.py:87]:
```suggestion
            if node_object.condition() is None or node_object.condition().evaluate(context):
                valid_composable_nodes.append(node_object)
```

**Review by ivanpauno — APPROVED** [review]
LGTM!

File: launch_ros/launch_ros/actions/composable_node_container.py (hunk 1 above)
Comment (adityapande-1995) [reply launch_ros/launch_ros/actions/composable_node_container.py]: Done https://github.com/ros2/launch_ros/pull/311/commits/658c7681e0e3d4234898da2e87514e3e6e8f873b

**Comment by ivanpauno** [comment]
@adityapande-1995 could you fix CI failures?

**Comment by adityapande-1995** [comment]
CI : 

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/16607/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/11191/)
* Windows [Build Status]

**Comment by adityapande-1995** [comment]
CI round 2 with flake8 fixes : 
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/16608/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/11192/)
* Windows [Build Status]

**Comment by adityapande-1995** [comment]
Got green CI, merging !

**Comment by sea-bass** [comment]
Any chance this can be backported to Humble? We just had a MoveIt user find an issue with something related to this change not being in Humble.

**Comment by mjcarroll** [comment]
> Any chance this can be backported to Humble? We just had a MoveIt user find an issue with something related to this change not being in Humble.

I don't see any reason why not after looking at it.

**Pull request merged**